scripts/Version.scala

```diff
@@ -1,55 +1,31 @@
 package besom.scripts
 
+import besom.codegen.Config
 import besom.model.SemanticVersion
 
 import scala.sys.exit
 import scala.util.Try
 import scala.util.matching.Regex
 
 object Version:
-  private val cwd               = besomDir
-  lazy val besomVersion: String = os.read(cwd / "version.txt").trim
-
-  private lazy val latestPackageVersions = fetchLatestPackageVersions()
+  private val cwd = Config.besomDir
 
   def main(args: String*): Unit =
+    given config: Config = Config()
 
     lazy val besomDependencyPattern: Regex =
-      ("""^(//> using (?:test\.)?(?:dep|lib|plugin) +"?\S+::besom-)([^"]+)("?)$""").r
-
-    val expectedFileNames = Vector("project.scala", "project-test.scala", "run.scala")
-
-    def projectFiles(path: os.Path = cwd): Map[os.Path, String] =
-      println(s"Searching for project files in $path\n")
-      os.walk(
-        path,
-        skip = (p: os.Path) => p.last == ".git" || p.last == ".out" || p.last == ".bsp" || p.last == ".scala-build" || p.last == ".idea"
-      ).filter(f => expectedFileNames.contains(f.last))
-        .map((f: os.Path) => f -> os.read(f))
-        .toMap
-
-    def changeVersion(version: String, newBesomVersion: String, packageVersion: String => Option[String] = _ => None): String =
-      lazy val coreShortVersion = SemanticVersion
-        .parseTolerant(besomVersion)
-        .fold(
-          e => throw Exception(s"Invalid besom version: ${besomVersion}", e),
-          _.copy(patch = 0).toShortString
-        )
-      version match
-        case s"$a:$b-core.$_" => s"$a:${packageVersion(a).getOrElse(b)}-core.$coreShortVersion"
-        case s"$a:$_"         => s"$a:$newBesomVersion"
-    end changeVersion
+      """^(//> using (?:test\.)?(?:dep|lib|plugin) +"?\S+::besom-)([^"]+)("?)$""".r
```

Comment:
is it possible to have a comment explaining this pattern a bit? prolly gpt can do this quite well

Reply:
I've added a comment explaining the intent of the regexp and te highlights. I didn't include a full explanation of the regexp symbol by symbol because I find it redundant.